Andrea jaqueline

chilimatic opened the pull request:
before it's to late ;)

chilimatic committed:
- merge of non angular jaqueline branch to andrea
- not fully tested

chilimatic committed:
-removed duplicate header send

chilimatic committed:
- unit test for config
- composer json with unitest

chilimatic committed:
- added db access
- fixed unittests
- extended gitignore file

chilimatic committed:
- minor fixes

andreas83 commented on app/lib/transformer/string/html/HashTransformer.php:
```diff
@@ -0,0 +1,26 @@
+     * pattern for search and replace
+     */
+    const PATTERN = '/(^|\s)#(\w*[a-zA-Z0-9öäü_-]+\w*)/';
+    const REPLACE_PATTEN = '\1<a href="http://www.dasmerkendienie.com/hash/\2">#\2</a>';
```
hostname should be part of config

chilimatic replied:
thx, it wasn't there at the time. fixed

andreas83 commented on app/lib/transformer/string/html/LinkTransformer.php:
```diff
@@ -0,0 +1,26 @@
+ *
+ * File: LinkTransformer.php
+ */
+class RedtubeTransformer implements IStringTransformer
```
class name is RedtubeTransformer while Filename is LinkTransformer ?

chilimatic replied:
thx.

andreas83 commented:
just as a quick feedback:
i highly appreciate the idea of having a request and response object, good job!
also the gulp stuff looks pretty promising, we should add the jsx transformation for react too...

Commit:
Merge branch 'andrea' into andrea-jaqueline

chilimatic committed:
- bugfix code review

Commit:
moved jquery, bootstrap dependencies from composer 2 bower

Commit:
Added highlightjs to bower

Commit:
added babel jsx transformer to gulp (some further work needs to be done here )

andreas83 commented:
i removed the jquery, bootstrap and highlight js part from composer.json 
in favor of bower...  
at moment the css and js stuff is included in footer.php, 
i would love to see a better approach, like require.js oder webpack. any recommendations ?

Commit:
added fontawesome

Commit:
typo bugfix

Commit:
Merge branch 'andrea-jaqueline' of https://github.com/andreas83/SocialNetwork into andrea-jaqueline

chilimatic commented:
i use require.js in my angular projects, but for my cycleJS experiments I have webpack which is still kinda confusing. And if I remember correctly one of the guys @ the meetup mentioned that if you use webpack you have to use it for everything. 

I do know requireJS so I can help you there. I personally (at least now) prefer the bundle approach instead of the lazy loading. The lazy loading seams cool but in the end the HTTP requests add up and mapping them in JS can create the same issues as just compiling / concating them to packages for specific areas. 

But maybe there are benefits, I'd have to study them. Atm I would use require JS and create "area"-bundles for example "frontend.bundle.min.js".

Commit:
added gulp-babel to package

Commit:
removed JSON_NUMERIC_CHECK, since this breaks quite a lot string functions from frontend

Commit:
critical

Commit:
Fixed some notices
Added Trending Hashtags
Fixed some error in auth token generation

Commit:
fixed typo in composer.json

Commit:
added babel jsx transformation to gulpfile 
moved ever frontend lib (jquery, bootstrap, textcomplete etc) to bower
next step would be to package everything via webpack or require.js

andreas83 commented:
I moved all frontend plugins (jquery, bootstrap, autocomplert) from composer to bower. 

Todo:
- [x] gulp file (react task jsx transformation)
- [x] reduce http calls - too many js css loaded via footer.php (maybe we can uglify it or/and use require.js)
- [x] cleanup (please remove all unused node dependencies)
- [x] testing

Right now it looks most of the stuff works, but since the react taks is broken iam not able todo further test..

chilimatic commented:
and which one are the unused node dependencies ? :D I can't read your mind ! :)

andreas83 commented:
i thought you know which dependencies are required since you wrote the package.json. 
i guess the sass part is not needed anymore?!
of cause we could move the sass parsing from php 2 gulp too.

{
  "name": "dmdn",
  "version": "0.0.1",
  "description": "dmdn socket IO chat",
  "dependencies": {
    "bower": "_",
    "sass": "_",
    "gulp": "_",
    "gulp-sass": "_",
    "gulp-minify-css": "_",
    "gulp-jshint": "_",
    "gulp-concat": "_",
    "gulp-uglify": "_",
    "gulp-livereload": "_",
    "gulp-cache" : "_",
    "gulp-notify" : "_",
    "gulp-rename" : "_",
    "gulp-babel" : "_",
    "del" : "_",
    "react-tools": "*"
  },
  "scripts": {
    "start" : ""
  }
}

chilimatic commented:
ahh i see, bower is still useful for specific packages like jquery and such, 
The rest is really just gulp, I will try to fix the gulpfile, unify the "build" and maybe explain it to you tonight for 5 minutes why I do what and how and you can take it over and change the things you don't like, or tell me what you want in another way.

andreas83 commented:
sounds good, lets have a quick chat tonight !

Commit:
removed sass
failed to write gulp task react

Commit:
added react preset

andreas83 commented:
looks like i do not get my gulp task "react" working properly. to be more specific, 
the generated code does not work like the generated code from cli tool.

**cli works fine**
babel --presets react -w Author.jsx Comments.jsx Stream.jsx Likes.jsx SearchBox.jsx Chat.jsx ShareBox.jsx Notifications.jsx InitStream.jsx --watch --out-file react.js

**gulp task**
gulp.task('react', function() {
    return gulp.src('public/js/jsx/*.jsx')
        .pipe(babel({"presets": ["react"]}))
        .pipe(concat('react.js'))
        .pipe(gulp.dest('public/js/'))
});

any clue whats wrong ?

andreas83 commented:
found the soloution (its the order)

Commit:
fixed order of gulpfile
we have now a working react task wohoo

Commit:
reduced http request aka moved all js files into one app.js

andreas83 commented:
i reduced the http request by moved every js into one big app.js 

what do you think ?

chilimatic commented:
sorry I was working till now, I would do the same thing maybe group it into specific bundles.
Btw a guy from Mobfox mentioned that you (everyone) should look out for the re-rendering in react because most developers forget to check if the component needs to be re-rendered. 

Do you stilll need my help ? otherwise I would just leave it to your capable hands

Commit:
removed functions.jsx
added uglify
fixed some typos

Commit:
added react also to app.js

chilimatic committed:
-> code formating (einmal mit leuten arbeiten die sich ein bisschen drum scheren ;) )

andreas83 commented:
re rendering ? 
for now the gulp task works for me, every javascript  (jquery, react, bootstrap etc) is now one uglified app.js. 

Pro
- we do not depend on any external source (will work on the intranet)

Con 
- Not sure if one big js file is better than 10 small ones.

thx for your help, i highly appreciate your work (in special the request, response part, as well the gulp stuff) 

next task for me will be testing and merge the whole thing, also i like to focus on documentation of the development stack. 
Lets talks about phing and unit test next time!

chilimatic commented:
1 big is less blocking than 10 small ones. So For me it always wins, bigger companies like google deliver 1 basic small inline package and load the bigger scripts afterwards. but that's a different topic :)

Commit:
moved css into one file

Commit:
removed php scss parser in favor of gulp scss task
added minified version

Commit:
removed due new gulp icons task

Commit:
removed in favor of new sass gulp task

Commit:
Merge branch 'andrea-jaqueline' of https://github.com/andreas83/SocialNetwork into andrea-jaqueline

andreas83 commented on composer.json:
```diff
@@ -7,11 +7,10 @@
-        "league/oauth2-google": "^1.0",
-        "textalk/websocket": "^1.2"
-
+        "league/oauth2-client"      : "^1.3",
```
:heart: :heart: :heart:
